### PR TITLE
darwin.builder: auto-login as the `builder` user

### PR DESCRIPTION
**Pull request by Gabriella439**
… as suggested by @NiklasGollenstede in:

https://github.com/NixOS/nixpkgs/pull/206951#issuecomment-1369020601

###### Things done

- Built on platform(s)

  - [x] aarch64-darwin
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by winterqt — CHANGES_REQUESTED** [review]
Might make sense to inline the reasoning, too (in the commit message)? IDK, LGTM otherwise.

**Comment by winterqt** [comment]
Thanks, just want to quickly sanity check it; can do so in ~18 hours, if you don't mind waiting. Thanks again!

File: nixos/modules/profiles/macos-builder.nix [hunk 1]
```diff
@@ -59,10 +59,14 @@ in
     trusted-users = [ "root" user ];
   };
 
-  services.openssh = {
-    enable = true;
+  services = {
+    getty.autologinUser = "builder";
```
Comment (raphaelr) [inline nixos/modules/profiles/macos-builder.nix:63]:
```suggestion
    getty.autologinUser = user;
```

**Review by winterqt — APPROVED** [review]
LGTM other than that single doc nit, PTAL.

**Pull request merged**